### PR TITLE
Chaos Labs Scope and Compensation Amendment

### PR DESCRIPTION
**Pull request by yonikesel**
Discussion: https://governance.aave.com/t/arfc-chaos-labs-scope-and-compensation-amendment/14407

### CODE REVIEW

**Review by sakulstra — DISMISSED** [review]

File: src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * command: make test-contract filter=AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816
+ */
+contract AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816_Test is
+  ProtocolV3TestBase
+{
+  AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816 internal proposal;
+
+  IERC20 public constant AUSDT = IERC20(AaveV2EthereumAssets.USDT_A_TOKEN);
+
+  ICollector public immutable AAVE_COLLECTOR = AaveV2Ethereum.COLLECTOR;
```
Comment (sakulstra) [inline src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol:24]:
```suggestion
  ICollector public constant AAVE_COLLECTOR = AaveV2Ethereum.COLLECTOR;
```

File: src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  address public constant CHAOS_LABS_TREASURY = 0xbC540e0729B732fb14afA240aA5A047aE9ba7dF0;
+
+  IStreamable public immutable STREAMABLE_AAVE_COLLECTOR =
+    IStreamable(address(AaveV2Ethereum.COLLECTOR));
```
Comment (sakulstra) [inline src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol:28]:
```suggestion
```

`IStreamable should no longer be needed. `ICollector` exposes all the methods you need-
https://github.com/bgd-labs/aave-address-book/blob/main/src/common/ICollector.sol#L173

File: src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  uint256 public constant actualAmountaUSDT = (STREAM_AMOUNT / STREAM_DURATION) * STREAM_DURATION;
+
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('mainnet'), 17926515);
+    proposal = new AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816();
+  }
+
+  function testProposalExecution() public {
+    // Capturing next Stream IDs before proposal is executed
+    uint256 nextCollectorStreamID = STREAMABLE_AAVE_COLLECTOR.getNextStreamId();
```
Comment (sakulstra) [inline src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol:42]:
```suggestion
    uint256 nextCollectorStreamID = AAVE_COLLECTOR.getNextStreamId();
```

File: src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      (
+        address senderUSDT,
+        address recipientUSDT,
+        uint256 depositUSDT,
+        address tokenAddressUSDT,
+        uint256 startTimeUSDT,
+        uint256 stopTimeUSDT,
+        uint256 remainingBalanceUSDT,
+
+      ) = STREAMABLE_AAVE_COLLECTOR.getStream(nextCollectorStreamID);
```
Comment (sakulstra) [inline src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol:58]:
```suggestion
      ) = AAVE_COLLECTOR.getStream(nextCollectorStreamID);
```

File: src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      assertEq(remainingBalanceUSDT, actualAmountaUSDT);
+    }
+
+    // Checking if Chaos can withdraw from streams
+    uint256 initialChaosUSDTBalance = AUSDT.balanceOf(CHAOS_LABS_TREASURY);
+    vm.startPrank(CHAOS_LABS_TREASURY);
+    vm.warp(block.timestamp + (STREAM_DURATION / 2));
+    uint256 firstPayment = actualAmountaUSDT / 2;
+
+    STREAMABLE_AAVE_COLLECTOR.withdrawFromStream(nextCollectorStreamID, firstPayment);
```
Comment (sakulstra) [inline src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol:74]:
```suggestion
    AAVE_COLLECTOR.withdrawFromStream(nextCollectorStreamID, firstPayment);
```

File: src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+    uint256 halfChaosUSDTBalance = AUSDT.balanceOf(CHAOS_LABS_TREASURY);
+    assertEq(initialChaosUSDTBalance, halfChaosUSDTBalance - (actualAmountaUSDT / 2));
+
+    vm.warp(block.timestamp + (STREAM_DURATION / 2) + 1 days);
+    uint256 midStreamChaosUSDTBalance = AUSDT.balanceOf(CHAOS_LABS_TREASURY);
+    uint256 secondPayment = actualAmountaUSDT - firstPayment;
+
+    STREAMABLE_AAVE_COLLECTOR.withdrawFromStream(nextCollectorStreamID, actualAmountaUSDT / 2);
```
Comment (sakulstra) [inline src/20230816_AaveV3_Eth_ChaosLabsScopeAndCompensationAmendment/AaveV3_Ethereum_ChaosLabsScopeAndCompensationAmendment_20230816.t.sol:83]:
```suggestion
    AAVE_COLLECTOR.withdrawFromStream(nextCollectorStreamID, actualAmountaUSDT / 2);
```

**Review by sakulstra — APPROVED** [review]

**Review by Zer0dot — APPROVED** [review]

**Pull request merged**